common/lib/client/realtimechannel.js

```diff
@@ -83,7 +86,24 @@ var RealtimeChannel = (function() {
 		if (this._decodingContext)
 			this._decodingContext.channelOptions = this.channelOptions;
 		if(this._shouldReattachToSetOptions(options)) {
-			this._attach(true, null, callback);
+			/* This does not just do _attach(true, null, callback) because that would put us
```

Comment:
Does this need a spec clarification?

Reply:
yeah, we should

Reply:
I've created https://github.com/ably/docs/issues/894 

@SimonWoolf, my only concern on a skim read of this code (sorry, I've not looked in great depth) is to ensure that you've considered that the `_shouldReattachToSetOptions` 'method' implementation returns `true` for both the `attaching` as well as the `attached` local states, per [RTL16a](https://docs.ably.io/client-lib-development-guide/features/#RTL16a). Would this code running when the channel is `attaching` have a different outcome?

Reply:
[moved](https://ably-real-time.slack.com/archives/C030C5YLY/p1591598178340200) to internal Slack channel